### PR TITLE
Lazy codegen change for xla

### PR DESCRIPTION
**Pull request by JackCaoG**
Codegen change to enable PyTorch/XLA to generate the first op in https://github.com/pytorch/xla/pull/3544.

@bdhirsh @wconstab

### CODE REVIEW

File: torch/csrc/lazy/core/shape_inference.h [hunk 1]
```diff
@@ -11,59 +11,59 @@
...
-TORCH_API std::vector<Shape> compute_shape_std(const at::Tensor & self, bool unbiased);
-TORCH_API std::vector<Shape> compute_shape_std(const at::Tensor & self, at::IntArrayRef dim, bool unbiased, bool keepdim);
-TORCH_API std::vector<Shape> compute_shape_std(const at::Tensor & self, at::OptionalIntArrayRef dim, c10::optional<int64_t> correction, bool keepdim);
-TORCH_API std::vector<Shape> compute_shape_sum(const at::Tensor & self, c10::optional<at::ScalarType> dtype);
-TORCH_API std::vector<Shape> compute_shape__to_copy(const at::Tensor & self, c10::optional<at::ScalarType> dtype, c10::optional<at::Layout> layout, c10::optional<at::Device> device, c10::optional<bool> pin_memory, bool non_blocking, c10::optional<at::MemoryFormat> memory_format);
-TORCH_API std::vector<Shape> compute_shape_trace(const at::Tensor & self);
-TORCH_API std::vector<Shape> compute_shape_zero_(at::Tensor & self);
-TORCH_API std::vector<Shape> compute_shape_narrow_copy(const at::Tensor & self, int64_t dim, int64_t start, c10::SymInt length);
+TORCH_API std::vector<torch::lazy::Shape> compute_shape__adaptive_avg_pool2d(const at::Tensor & self, at::IntArrayRef output_size);
```
Comment (wconstab) [inline torch/csrc/lazy/core/shape_inference.h:14]: haha. ok, we may have a namespace war on our hands

Comment (wconstab) [reply torch/csrc/lazy/core/shape_inference.h:14]: (this is ok to change to lazy:: IMO.  but.. i also worry a little bit that people will not realize it breaks something when they change it back as a cleanup)

Comment (JackCaoG) [reply torch/csrc/lazy/core/shape_inference.h:14]: I think as long as that pr trigger a xla test they will see xla build fail.

**Comment by JackCaoG** [comment]
dumb question, where can I find the linter instruction for pytorch 😄 ?

**Comment by wconstab** [comment]
> dumb question, where can I find the linter instruction for pytorch 😄 ?

https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md#local-linting
tl;dr you just need to `make setup_lint` once and then `make quicklint` usually

**Review by wconstab — COMMENTED** [review]
LGTM, i'll let nick approve

**Comment by JackCaoG** [comment]
CI Failures are related to `lu_solve` which I think is irrelevant to this pr. @Krovatkin I think this pr is ready to be reviewed. This pr can also just go in without xla pr.

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Comment by JackCaoG** [comment]
Let me resolve the conflict and try to land this.
